src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1691,30 +1678,27 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             }
 
             _terminal->SetSearchHighlights(_searcher.Results());
-            _terminal->SetSearchHighlightFocused(_searcher.CurrentMatch());
+            _renderer->TriggerSearchHighlight(oldResults);
         }
-        else
+        else if (!reset)

```

Comment:
This allows us to ensure that we don't accidentally call `FindNext()` just because the `OutputIdle` event got triggered, which results in a `Search()` call.